### ISSUE TITLE
inception_thoughts doesn't work in x86_64 Mac OS X

### ISSUE THREAD

**Issue by dgladkov** [karthick18/inception#2]
inception_thoughts code throws "illegal hardware instruction" in x86_64 Mac OS X unless you compile the program with -m32

**Comment by karthick18** [comment]
Yeah. I don't have a MAC x86_64 to test since x86_64 syscall interface should be slightly different for OSX 64 then its for 64 bit. Also the registers could be different for the arguments pushed into the stack before a write syscall unlike 32 bit. I will see what I can do for x86_64 MAC when I get time, since I tested this on a MAC VM (vmplayer) running on linux.
